src/sentry/api/endpoints/internal/rpc.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from django.conf import settings
+from OpenSSL.crypto import X509Name
+from rest_framework import permissions
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.authentication import MutualTlsAuthentication
+from sentry.api.base import Endpoint, control_silo_endpoint, region_silo_endpoint
+from sentry.services.hybrid_cloud.rpc import ServiceEndpoint, endpoint_registry
+from sentry.services.hybrid_cloud.rpc.endpoints import MethodEndpoint
+from sentry.silo import SiloLimit, SiloMode
+
+
+class SiloPermission(permissions.BasePermission):
+    def has_permission(self, request: Request, view: RpcEndpoint):
+        auth: X509Name = request.auth
+        if not isinstance(request.auth, X509Name):
+            common_name: str = auth.commonName
+            is_control_silo = common_name == settings.CONTROL_SILO_COMMON_NAME
+
+            # Region silo RPCs cannot communicate with each other; they may only be
+            # contacted on behalf of the control silo.
+            if view.silo_mode() == SiloMode.REGION:
```

Comment:
The idea is that the control silo can call any region silo, and the control silo can be called by any silo, but region silos cannot call other region silos.